### PR TITLE
code health SHA2

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

 Update sha256t and sha512t to use the test infrastructure

### CODE REVIEW

File: test/sha256t.c [hunk 1]
```diff
@@ -56,122 +55,118 @@ static const unsigned char addenum_3[SHA224_DIGEST_LENGTH] = {
...
+        goto end;
+    for (i = 0; i < 1000000; i += 288)
+        if (!TEST_true(EVP_DigestUpdate(evp, updstr,
+                                        (1000000 - i) < 288 ? 1000000 - i
+                                                            : 288)))
+            goto end;
+    if (!TEST_true(EVP_DigestFinal_ex(evp, md, NULL)))
+            goto end;
+
+    if (!TEST_mem_eq(md, sizeof(md), app_b3, sizeof(app_b3)))
```
Comment (mattcaswell) [inline test/sha256t.c]: Make a single if

File: test/sha256t.c [hunk 2]
```diff
@@ -56,122 +55,118 @@ static const unsigned char addenum_3[SHA224_DIGEST_LENGTH] = {
...
+    if (!TEST_true(EVP_DigestInit_ex(evp, EVP_sha224(), NULL)))
+        goto end;
+    for (i = 0; i < 1000000; i += 64)
+        if (!TEST_true(EVP_DigestUpdate(evp, updstr,
+                                        (1000000 - i) < 64 ? 1000000 - i : 64)))
+            goto end;
+    if (!TEST_true(EVP_DigestFinal_ex(evp, md, NULL)))
+        goto end;
+
+    if (!TEST_mem_eq(md, sizeof(md), addenum_3, sizeof(addenum_3)))
```
Comment (mattcaswell) [inline test/sha256t.c]: Single if

File: test/sha512t.c [hunk 3]
```diff
@@ -75,125 +73,123 @@ static const unsigned char app_d3[SHA384_DIGEST_LENGTH] = {
...
+    if (!TEST_true(EVP_DigestInit_ex(evp, EVP_sha512(), NULL)))
+        goto end;
+    for (i = 0; i < 1000000; i += 288)
+        if (!TEST_true(EVP_DigestUpdate(evp, updstr,
+                                        (1000000 - i) < 288 ? 1000000 - i
+                                                            : 288)))
+            goto end;
+    if (!TEST_true(EVP_DigestFinal_ex(evp, md, NULL)))
+        goto end;
     EVP_MD_CTX_reset(evp);
```
Comment (mattcaswell) [inline test/sha512t.c]: This line is now redundant and can be removed.

File: test/sha512t.c [hunk 4]
```diff
@@ -75,125 +73,123 @@ static const unsigned char app_d3[SHA384_DIGEST_LENGTH] = {
...
+    if (!TEST_true(EVP_DigestInit_ex(evp, EVP_sha384(), NULL)))
+        goto end;
+    for (i = 0; i < 1000000; i += 64)
+        if (!TEST_true(EVP_DigestUpdate(evp, updstr,
+                                        (1000000 - i) < 64 ? 1000000 - i : 64)))
+            goto end;
+    if (!TEST_true(EVP_DigestFinal_ex(evp, md, NULL)))
+        goto end;
+
+    if (!TEST_mem_eq(md, sizeof(md), app_d3, sizeof(app_d3)))
```
Comment (mattcaswell) [inline test/sha512t.c]: Make a single if

**Comment by pauli-oracle** [comment]
Made lots of single ifs.

**Comment by richsalz** [comment]
+1

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**